CrowServe/CrowServe.cpp

```diff
@@ -17,90 +17,61 @@ void deinit_sockets(){
 #endif
 }
 
-bool Socket::try_init() {  
+void Socket::try_init(std::stop_token stop_token) {
+    m_mutex.lock();
```

Comment:
Typicially should use a scope (so empty bracket block) and start it with lock guard. Otherwise this code is not exception safe, and also is less clean from cpp standpoint.